sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +112,31 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propName: String, jarName: String, 
+      complainFun: (Boolean, String) => Unit): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val rapidsJarURLs = classloader.getResources(propName).asScala.toSet
+    lazy val rapidsJars = rapidsJarURLs.map(_.toString.split("!").head).mkString(",")
+    lazy val rapidsJarsVers = rapidsJarURLs.map { 
+      url => scala.io.Source.fromInputStream(url.openStream()).mkString("") 
+    }.mkString(",")
+    complainFun(rapidsJarURLs.size <= 1,
+        s"Multiple $jarName jars found in the classpath: $rapidsJars, please make sure there " +
+        s"is only one $jarName jar in the classpath. If it is impossible to fix the classpath " + 
+        s"you can suppress the error by setting ${RapidsConf.ALLOW_MULTIPLE_JARS.key} to true. " + 
+        s"Version info: \n$rapidsJarsVers")
+  }
+
+  def detectMultipleJars(conf: RapidsConf): Unit = {
+    val complainFun: (Boolean, String) => Unit = conf.allowMultipleJars match {
```

Comment:
NIT: If all the props share the same complain function, we can remove it and implement this logic inside the `detectMultipleJar` method.

Reply:
Good suggestion, done.